config/src/trusted_peers.rs

```diff
@@ -82,6 +110,29 @@ where
     Ok(from_encoded_string(encoded_key))
 }
 
+pub fn deserialize_nextgen_key<'de, D, K>(deserializer: D) -> Result<K, D::Error>
```

Comment:
This would eventually require renaming, when we move our algorithms from `nextgen`. Actually with this and similar refactorings, it's not considered nextgen anymore, right?

Reply:
Made them squat the main name, renamed X25519 to _legacy_